src/server/scripts/Outland/zone_hellfire_peninsula.cpp

```diff
@@ -1039,6 +1041,259 @@ class npc_magister_aledis : public CreatureScript
+            {
+                switch (eventId)
+                {
+                    case EVENT_INFERNAL_RAIN_CAST:
+                    {
+                        std::list<Creature*> InfernalRainList;
+                        Trinity::AllCreaturesOfEntryInRange checker(me, NPC_INFERNAL_RAIN, 200.0f);
+                        Trinity::CreatureListSearcher<Trinity::AllCreaturesOfEntryInRange> searcher(me, InfernalRainList, checker);
+                        Cell::VisitAllObjects(me, searcher, 200.0f);
+
+                        if (InfernalRainList.empty())
+                            return;
+
+                        if (Creature* random = Trinity::Containers::SelectRandomContainerElement(InfernalRainList))
+                        {
+                            if (random->isMoving() && random->GetPositionZ() < 118.0f)
+                            {
+                                me->CastSpell(random, SPELL_INFERNAL_RAIN, true);
+                            }
+                        }
+
+                        _events.ScheduleEvent(EVENT_INFERNAL_RAIN_CAST, Seconds(1), Seconds(2));
+                        break;
+                    }
+                    case EVENT_INFERNAL_RAIN_STOP:
+                    {
+                        _events.CancelEvent(EVENT_INFERNAL_RAIN_CAST);
+
+                        Creature* Watch_Commander_Leonus = me->FindNearestCreature(NPC_WATCH_COMMANDER_LEONUS, 200);
+                        if (Watch_Commander_Leonus)
+                            Watch_Commander_Leonus->AI()->SetData(1, 1);
+
+                        break;
+                    }
+                }
+            }
+        }
+
+    private:
+        EventMap _events;
+    };
+
+    CreatureAI* GetAI(Creature* creature) const override
+    {
+        return new npc_infernal_rain_hellfireAI(creature);
+    }
+};
+
+enum fear_controller
+{
+    EVENT_FEAR_CAST = 1,
+    EVENT_FEAR_STOP = 2,
+    SPELL_FEAR = 33815 // Note: this spell isn't in DBC but found in sniffs
+};
+
+class npc_fear_controller : public CreatureScript
+{
+public:
+    npc_fear_controller() : CreatureScript("npc_fear_controller") { }
+
+    struct npc_fear_controllerAI : public ScriptedAI
+    {
+        npc_fear_controllerAI(Creature* creature) : ScriptedAI(creature) { }
+
+        void SetData(uint32 /*type*/, uint32 data) override
+        {
+            switch (data)
+            {
+            case 1:
+                _events.ScheduleEvent(EVENT_FEAR_CAST, Seconds(1));
+                _events.ScheduleEvent(EVENT_FEAR_STOP, Seconds(60));
+                break;
+            default:
+                break;
+            }
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                    case EVENT_FEAR_CAST:
+                    {
```

Comment:
brackets not needed in both cases.
You just need it if you declare some variable inside of case